### PR TITLE
[bugfix] QNodes can no longer return empty iterables

### PR DESCRIPTION
**Pull request by timmysilv**
**Context:**
QNodes are supposed to always return 1 or more measurement processes, and nothing else. Before this change, if a QNode returned an empty iterable, it would decree that "all of the returned values are of some measurement type", which would true! All zero of them, that is, so it would then succeed and return the empty iterable. However, QNodes are supposed to return at least 1 measurement as previously stated, so this should be fixed.

**Description of the Change:**
- Update `QNode.construct` to make sure the qfunc output is truthy (it already otherwise ensured that all return values are measurements)
- Update a `qfunc_transform` docstring example because it would fail after this change (which is how I even found this bug)

**Benefits:**
The QNode fails as expected if it doesn't return any measurement. This will be useful because otherwise, users might accidentally execute QNodes thinking something meaningful is being done when really, nothing is. That would be especially bad if they didn't notice this while running some code on pay-per-use compute resources.

**Possible Drawbacks:**
Perhaps some users have found this undesired behaviour and have grown to expect it? I think it's good to merge.

### CODE REVIEW

File: pennylane/transforms/qfunc_transforms.py [hunk 1]
```diff
@@ -182,10 +182,10 @@ def internal_wrapper(*args, **kwargs):
...
         tape = tape_transform(tape, *transform_args, **transform_kwargs)
 
-        if len(tape.measurements) == 1:
-            return tape.measurements[0]
-
-        return tape.measurements
+        num_measurements = len(tape.measurements)
+        if num_measurements == 0:
+            return None
+        return tape.measurements[0] if num_measurements == 1 else tape.measurements
```
Comment (albi3ro) [inline pennylane/transforms/qfunc_transforms.py:188]: We should have a unit test for this behaviour too.

File: tests/test_qnode.py [hunk 2]
```diff
@@ -680,6 +680,19 @@ def func(x, y):
...
+            qml.CNOT(wires=[0, 1])
+            return []
+
+        qn = QNode(func, dev)
+
+        with pytest.raises(
+            qml.QuantumFunctionError, match="must return either a single measurement"
+        ):
+            qn(5, 1)
+
```
Comment (mudit2812) [inline tests/test_qnode.py:695]: Since the test has been updated, I think it should be linted and added to `tests_passing_pylint.txt`

Comment (timmysilv) [reply tests/test_qnode.py:695]: there are ~80 violations, I don't think it's worth my time tbh 😄

File: pennylane/transforms/qfunc_transforms.py (hunk 1 above)
Comment (timmysilv) [reply pennylane/transforms/qfunc_transforms.py:188]: added

**Review by mudit2812 — APPROVED** [review]
Looks good 🎉

File: tests/test_qnode.py (hunk 2 above)
Comment (albi3ro) [reply tests/test_qnode.py:695]: At that many, agree.

**Review by albi3ro — APPROVED** [review]
Looks good to me.

**Pull request merged**